examples/stateful-precompile/src/main.rs

```diff
@@ -63,58 +63,41 @@ pub struct MyEvmFactory {
 }
 
 impl EvmFactory<EvmEnv> for MyEvmFactory {
-    type Evm<DB: Database, I> = EthEvm<DB, I>;
+    type Evm<DB: Database, I: Inspector<EthEvmContext<DB>, EthInterpreter>> = WrappedEthEvm<DB, I>;
     type Tx = TxEnv;
     type Error<DBError: core::error::Error + Send + Sync + 'static> = EVMError<DBError>;
     type HaltReason = HaltReason;
     type Context<DB: Database> = EthEvmContext<DB>;
 
-    fn create_evm<'a, DB: Database + 'a>(
-        &self,
-        db: DB,
-        input: EvmEnv,
-    ) -> Self::Evm<DB, NoOpInspector> {
-        let cfg_env_with_handler_cfg = CfgEnvWithHandlerCfg {
-            cfg_env: input.cfg_env,
-            handler_cfg: HandlerCfg::new(input.spec),
-        };
-
+    fn create_evm<DB: Database>(&self, db: DB, input: EvmEnv) -> Self::Evm<DB, NoOpInspector> {
         let new_cache = self.precompile_cache.clone();
-        EvmBuilder::default()
+
+        let evm = Context::mainnet()
             .with_db(db)
-            .with_cfg_env_with_handler_cfg(cfg_env_with_handler_cfg)
-            .with_block_env(input.block_env)
-            // add additional precompiles
-            .append_handler_register_box(Box::new(move |handler| {
-                MyEvmConfig::set_precompiles(handler, new_cache.clone())
-            }))
-            .build()
-            .into()
+            .with_cfg(input.cfg_env)
+            .with_block(input.block_env)
+            .build_mainnet()
+            .with_precompiles(WrappedPrecompile::new(EthPrecompiles::default(), new_cache));
+
+        EthEvm::new(evm)
     }
 
-    fn create_evm_with_inspector<'a, DB: Database + 'a, I: GetInspector<DB> + 'a>(
+    fn create_evm_with_inspector<DB: Database, I: Inspector<Self::Context<DB>, EthInterpreter>>(
         &self,
         db: DB,
         input: EvmEnv,
         inspector: I,
-    ) -> Self::Evm<'a, DB, I> {
-        let cfg_env_with_handler_cfg = CfgEnvWithHandlerCfg {
-            cfg_env: input.cfg_env,
-            handler_cfg: HandlerCfg::new(input.spec),
-        };
+    ) -> Self::Evm<DB, I> {
         let new_cache = self.precompile_cache.clone();
-        EvmBuilder::default()
+
+        let evm = Context::mainnet()
             .with_db(db)
-            .with_external_context(inspector)
-            .with_cfg_env_with_handler_cfg(cfg_env_with_handler_cfg)
-            .with_block_env(input.block_env)
-            // add additional precompiles
-            .append_handler_register_box(Box::new(move |handler| {
-                MyEvmConfig::set_precompiles(handler, new_cache.clone())
-            }))
-            .append_handler_register(inspector_handle_register)
-            .build()
-            .into()
+            .with_cfg(input.cfg_env)
+            .with_block(input.block_env)
+            .build_mainnet_with_inspector(inspector)
+            .with_precompiles(WrappedPrecompile::new(EthPrecompiles::default(), new_cache));
+
+        EthEvm::new(evm)
```

Comment:
like this should just be `self.create_evm(db, input).with_insp(inspector)`

Reply:
Hm maybe it is even possible now, let me check

Reply:
Yep, it is possible, have change the code